fsspec/caching.py

```diff
@@ -510,33 +576,37 @@ def _fetch(self, start, stop):
         return out + super()._fetch(start, stop)
 
 
-class UpdatableLRU:
+class UpdatableLRU(Generic[P, T]):
```

Comment:
Can we describe what P and T do here?

Reply:
Sure!

`T` is a type variable that represents the output of the function. `P` is a special `ParamSpec` that represents the function's signature (`*args` and `**kwargs`). That lets us re-use the exact signature of what function that we're caching here